proof/mint.go

```diff
@@ -259,21 +263,22 @@ func committedProofs(baseProof *Proof, taprootAssetRoot *commitment.TapCommitmen
 		// transition proofs.
 		assetProof.GenesisReveal = &newAsset.Genesis
 
-		// We also need to reveal the group key.
+		// If the asset has a group key, we only need to populate the
+		// group key reveal if the asset is the group anchor.
 		if newAsset.GroupKey != nil {
 			groupKey := newAsset.GroupKey
 
-			// All zero tapscript root means there is none.
-			var tapscriptRoot []byte
-			if groupKey.TapscriptRoot != [32]byte{} {
-				tapscriptRoot = groupKey.TapscriptRoot[:]
-			}
-
-			assetProof.GroupKeyReveal = &asset.GroupKeyReveal{
-				RawKey: asset.ToSerialized(
-					groupKey.RawKey.PubKey,
-				),
-				TapscriptRoot: tapscriptRoot,
+			err := anchorVerifier(
+				&newAsset.Genesis, groupKey,
+			)
+			if err == nil {
```

Comment:
So what if the `err` _is not_ `nil`? Should we add a return and then do the below non-indented?

Reply:
We don't want to return nor skip the rest of this logic, since we still need to store the proof.